src/Symfony/Component/ExpressionLanguage/Node/NullCoalesceNode.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ExpressionLanguage\Node;
+
+use Symfony\Component\ExpressionLanguage\Compiler;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @internal
+ */
+class NullCoalesceNode extends Node
+{
+    public function __construct(Node $expr1, Node $expr2)
+    {
+        parent::__construct(['expr1' => $expr1, 'expr2' => $expr2]);
+    }
+
+    public function compile(Compiler $compiler)
+    {
+        $compiler
+            ->raw('((')
+            ->compile($this->nodes['expr1'])
+            ->raw(') ?? (')
+            ->compile($this->nodes['expr2'])
+            ->raw('))')
+        ;
+    }
+
+    public function evaluate(array $functions, array $values)
+    {
+        if ($this->nodes['expr1'] instanceof GetAttrNode) {
+            $this->nodes['expr1']->isNullCoalesce = true;
+        }
+        
+        $expr1 = $this->nodes['expr1']->evaluate($functions, $values);
+        $expr2 = $this->nodes['expr2']->evaluate($functions, $values);
```

Comment:
the second expression should not be evaluated if the first one does not evaluate to `null`, to match the behavior of the `??` operator in PHP (and in the compiled mode)

Reply:
Just did the required update.. there will be no need for the intermediary variables to evaluate both expressions at the same time. Instead, we can place them directly in both-sides of the PHP's `??` native operator.